composer/utils/object_store/oci_object_store.py

```diff
@@ -0,0 +1,104 @@
+# Copyright 2022 MosaicML Composer authors
+# SPDX-License-Identifier: Apache-2.0
+
+"""S3-Compatible object store."""
+
+from __future__ import annotations
+
+import os
+import pathlib
+import uuid
+from typing import Callable, Optional, Union
+
+from composer.utils.import_helpers import MissingConditionalImportError
+from composer.utils.object_store.object_store import ObjectStore
+
+__all__ = ['OCIObjectStore']
+
+
+class OCIObjectStore(ObjectStore):
+    """Utility for uploading to and downloading from an OCI bucket.
+
+    Args:
+        bucket (str): The bucket name.
+        prefix (str): A path prefix such as `folder/subfolder/` to prepend to object names. Defaults to ''.
+    """
+
+    def __init__(
+        self,
+        bucket: str,
+        prefix: str = '',
+    ) -> None:
+        try:
+            import oci
+        except ImportError as e:
+            raise MissingConditionalImportError(conda_package='oci',
+                                                extra_deps_group='oci',
+                                                conda_channel='conda-forge') from e
+
+        # Format paths
+        self.bucket = bucket.strip('/')
+        self.prefix = prefix.strip('/')
+        if self.prefix:
+            self.prefix += '/'
+
+        if 'OCI_CONFIG_FILE' in os.environ:
+            config = oci.config.from_file(os.environ['OCI_CONFIG_FILE'])
+        else:
+            config = oci.config.from_file()
+
+        self.client = oci.object_storage.ObjectStorageClient(config=config,
+                                                             retry_strategy=oci.retry.DEFAULT_RETRY_STRATEGY)
+        self.namespace = self.client.get_namespace().data
+        self.upload_manager = oci.object_storage.UploadManager(self.client)
+
+    def get_uri(self, object_name: str) -> str:
+        return f'oci://{self.bucket}/{object_name}'
+
+    def get_object_size(self, object_name: str) -> int:
+        objects = self.client.list_objects(self.namespace, self.bucket, prefix=object_name, fields='size').data.objects
```

Comment:
Is there no direct get object api?

Reply:
can you use the same api is in download object? and then you don't need to raise your own error, can just rely on oci's error if object isn't found

Reply:
there is, but UploadManager handles large files better. see @karan6181's [comment ](https://github.com/mosaicml/composer/pull/1774#discussion_r1037832500)

Reply:
actually, one can use get_object. It was just more convoluted to get the size from get_object, but I figured it out. fixed